src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/InteropServices/DisableRuntimeMarshalling.cs

```diff
@@ -0,0 +1,354 @@
+                {
+                    bool canTransformToDisabledMarshallingEquivalent = CanTransformToDisabledMarshallingEquivalent(invocation);
+                    context.ReportDiagnostic(invocation.CreateDiagnostic(
+                        MethodUsesRuntimeMarshallingEvenWhenMarshallingDisabled,
+                        ImmutableDictionary.Create<string, string?>().Add(CanConvertToDisabledMarshallingEquivalentKey, canTransformToDisabledMarshallingEquivalent ? "true" : null),
+                        invocation.TargetMethod.ToDisplayString()));
+                }
+
+                static bool CanTransformToDisabledMarshallingEquivalent(IInvocationOperation invocation)
+                {
+                    return invocation.TargetMethod.Name switch
+                    {
+                        "OffsetOf" => false,
+                        "SizeOf" => invocation.TargetMethod.IsGenericMethod || (invocation.Arguments.Length > 0 && invocation.Arguments[0].Value is ITypeOfOperation { TypeOperand.IsUnmanagedType: true }),
+                        "StructureToPtr" => invocation.Arguments.Length > 0 && invocation.Arguments[0].Value is { Type.IsUnmanagedType: true },
+                        "PtrToStructure" => invocation.Type is not null,
+                        _ => false
+                    };
+                }
+            }
+
+            public void AnalyzeFunctionPointerCall(OperationAnalysisContext context)
+            {
+                var functionPointerInvocation = IFunctionPointerInvocationOperationWrapper.FromOperation(context.Operation);
+
+                if (functionPointerInvocation.GetFunctionPointerSignature().CallingConvention() == System.Reflection.Metadata.SignatureCallingConvention.Default)
+                {
+                    return;
+                }
+
+                AnalyzeMethodSignature(context.ReportDiagnostic, functionPointerInvocation.GetFunctionPointerSignature(), ImmutableArray.Create(functionPointerInvocation.WrappedOperation.Syntax.GetLocation()));
+            }
+
+            public void AnalyzeType(SymbolAnalysisContext context)
+            {
+                Debug.Assert(_unmanagedFunctionPointerAttribute is not null);
+                INamedTypeSymbol type = (INamedTypeSymbol)context.Symbol;
+                if (type.TypeKind != TypeKind.Delegate || !type.HasAttribute(_unmanagedFunctionPointerAttribute))
+                {
+                    return;
+                }
+
+                AnalyzeMethodSignature(context.ReportDiagnostic, type.DelegateInvokeMethod);
+            }
+
+            public void AnalyzeMethod(SymbolAnalysisContext context)
+            {
+                IMethodSymbol method = (IMethodSymbol)context.Symbol;
+
+                DllImportData? dllImportData = method.GetDllImportData();
+                if (dllImportData is null)
+                {
+                    return;
+                }
```

Comment:
Why is the return? There are other checks that don't depend on `dllImportData` below?

Reply:
This analyzer doesn't apply to any non-DllImported methods.